dagshub/data_engine/model/datasource.py

```diff
@@ -232,10 +232,19 @@ def delete_source(self, force: bool = False):
                 return
         self.source.client.delete_datasource(self)
 
-    def generate_datapoints(self):
+    def scan_source(self):
         """
         This function fires a call to the backend to rescan the datapoints.
-        Call this function whenever you updated/new files and want the changes to show up in the datasource metadata
+        Call this function whenever you uploaded new files and want the new files to appear when querying the datasource,
+        Or if you changed existing file contents and want their metadata to be updated automatically.
+
+        Notes about automatically scanned metadata:
+        1. Only new datapoints (files) will be added.
+           If files were removed from the source, their metadata will still remain,
+           and they will still be returned from queries on the datasource.
+           An API to actively remove metadata will be available soon.
+        2. Some metadata fields will be automatically scanned and updated by DagsHub based on this scan -
+           currently only "size" (bytes), but more metadata fields will be added automatically in the future.
```

Comment:
same. I'm not sure client docs are the best place for this kind of info in general

Reply:
```suggestion
           the list of automatic metadata fields is growing frequently!
```

Reply:
Honestly it's not great we don't have an official doc for this, we can lead to unexpected results.
We must have some reserved prefix or other convention for the fields we use, to avoid collisions with user fields

Reply:
valid point about reserved fields, yes